spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/kafka/KafkaItemReaderAutoConfiguration.java

```diff
@@ -0,0 +1,90 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.batch.autoconfigure.kafka;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Properties;
+
+import org.springframework.batch.item.kafka.KafkaItemReader;
+import org.springframework.batch.item.kafka.builder.KafkaItemReaderBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.batch.BatchAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+/**
+ *
+ * Autconfiguration for a {@code KafkaItemReader}.
+ *
+ * @author Glenn Renfro
+ * @since 2.3
+ */
+@Configuration
+@EnableConfigurationProperties({ KafkaProperties.class, KafkaItemReaderProperties.class })
+@AutoConfigureAfter(BatchAutoConfiguration.class)
+public class KafkaItemReaderAutoConfiguration {
+
+	@Autowired
+	private KafkaProperties kafkaProperties;
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnProperty(prefix = "spring.batch.job.kafkaitemreader", name = "name")
+	public KafkaItemReader<Object, Map<Object, Object>> kafkaItemReader(
+			KafkaItemReaderProperties kafkaItemReaderProperties) {
+		Properties consumerProperties = new Properties();
+		consumerProperties.putAll(this.kafkaProperties.getConsumer().buildProperties());
+		validateProperties(kafkaItemReaderProperties);
+		if (kafkaItemReaderProperties.getPartitions() == null
+				|| kafkaItemReaderProperties.getPartitions().size() == 0) {
+			kafkaItemReaderProperties.setPartitions(new ArrayList<>(1));
+			kafkaItemReaderProperties.getPartitions().add(0);
+		}
+		return new KafkaItemReaderBuilder<Object, Map<Object, Object>>()
+				.partitions(kafkaItemReaderProperties.getPartitions())
+				.consumerProperties(consumerProperties)
+				.name(kafkaItemReaderProperties.getName())
+				.pollTimeout(Duration
+						.ofSeconds(kafkaItemReaderProperties.getPollTimeOutInSeconds()))
+				.saveState(true).topic(kafkaItemReaderProperties.getTopic()).build();
```

Comment:
`saveState` is hard-coded to `true` here. Is this intended? I think it should be configurable like other readers.